src/main/webapp/scripts/fetchData.js

```diff
@@ -33,18 +44,45 @@ function fetchSpeciesData(name) {
 
         // Update description entry
         var notes = speciesData.wikipediaNotes;
-        descriptionContainer.innerText = notes == null ? "N/A" : notes.substr(0, notes.indexOf('['));
+        switch(notes) {
+            case null:
+                descriptionContainer.innerText = "N/A";
+                break;
+            case "":
+                descriptionContainer.innerText = "N/A";
+                break;
+            default:
+                descriptionContainer.innerText = notes.substr(0, notes.indexOf('['));
+        }
         citationsContainer.innerText = speciesData.citationLink;
 
         // Update image source
         img.src = speciesData.imageLink;
 
         // Manipulate pixelation value based on species population
         pixelSlider.max = img.width * img.height;
-
         var pop = speciesData.population;
-        pop = pop == null ? pixelSlider.max : pop.substr(0, pop.indexOf('['));
+        switch(true) {
+            case (pop == null):
+                pop = pixelSlider.max;
+                break;
+            case (pop.indexOf('–') > 0):
+                // Pixelation population is currently set to the lower bound if a range is given.
+                pop = pop.substr(0, pop.indexOf('–'));
+                break;
+            default:
+                break;
+        }
```

Comment:
Sorry about all the ternary operations, these are a lot clearer 😅

Reply:
Haha no worries! I would've liked to keep them since they're so concise but it started getting messy with the multiple conditions.